docs/recipes/LocalFirstDataWithPowerSync.md

```diff
@@ -0,0 +1,741 @@
+
+### Connecting to Your Database
+
+To tell PowerSync how to connect to the database we'll create a `PowerSyncBackendConnector` that
+has methods for fetching credentials and uploading data.
+
+#### PowerSyncBackendConnector Interface
+
+The Supabase Connector needs to implement the `PowerSyncBackendConnector` interface, ensuring it can seamlessly
+communicate with PowerSync for data synchronization. 
+
+The interface requires two methods:
+
+```ts
+// fetches credentials for connecting to the backend
+fetchCredentials: () => Promise<PowerSyncCredentials | null>;
+
+// a method powersync will call to upload data to the backend
+uploadData: (database: AbstractPowerSyncDatabase) => Promise<void>;
+```
+
+#### Implementing a Database Connector for Supabase
+
+We initialize the Supabase client with session persistence and custom storage for persisting keys. 
+
+Persisting the Supabase session is essential for maintaining user sessions across app restarts. Unlike web environments
+where `localStorage` is available, React Native requires a different approach for secure storage.
+
+You'll need to implement a simple key-value store to persist the session token and user ID. There are many options
+available for secure storage in React Native, such as `expo-secure-storage`, `react-native-keychain`, or `react-native-secure-key-store`.
+
+:::tip
+The`react-native-supabase-todolist` demo from PowerSync has a [simple implementation of a kv-store](https://github.com/powersync-ja/powersync-js/blob/main/demos/react-native-supabase-todolist/library/storage/KVStorage.ts)
+based on `expo-secure-storage` that works well for this.
+
+Note that you'll need to install `expo-secure-storage` to use this implementation in your app.
+:::
+
+
+
+```ts
+// app/services/database/supabase.ts
+import {
+    AbstractPowerSyncDatabase,
+    PowerSyncBackendConnector,
+    UpdateType
+} from "@journeyapps/powersync-sdk-react-native";
+import {createClient} from "@supabase/supabase-js";
+import Config from "app/config";
+import {KVStorage} from "./kv-storage";
+
+const kvStorage = new KVStorage();
+
+const client = createClient(Config.supabaseUrl, Config.supabaseAnonKey, {
+    auth: {
+        persistSession: true,
+        storage: kvStorage,
+    },
+});
+
+/**
+ * Called by PowerSync to fetch the credentials needed to connect to the backend.
+ * Here we use the Supabase client to get the session token and user ID.
+ */
+async function fetchCredentials() {
+    // NOTE: For client.auth.getSession() to work, the user needs to be signed in
+    const {data: {session}, error} = await client.auth.getSession();
+    if (!session || error) throw new Error(`Supabase credentials error: ${error?.message}`);
+
+    return {
+        client,
+        endpoint: Config.powersyncUrl,
+        token: session.access_token,
+        expiresAt: new Date(session.expires_at * 1000),
+        userID: session.user.id,
+    };
```

Comment:
Hrm  -- this I basically lifted from their demo. I get the sense that they are Java devs because they do a lot of control flow with exceptions.

I'll add a bit to return null. May as well implement it correctly.